### PR TITLE
feat: activity

### PR DESCRIPTION
**Pull request by LeifXu**
# Pull Request

## Related Issue

Closes FW-0000

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/app/(developers)/components/MethodButton.tsx [hunk 1]
```diff
@@ -52,6 +52,31 @@ export function MethodButton({ item }: Props) {
...
+                case SupportedMethod.BIND_WALLET: {
+                    const walletAddress = await fireflyBridgeProvider.request(SupportedMethod.BIND_WALLET, {
+                        type: Network.EVM,
+                    });
+                    enqueueInfoMessage(`Wallet Address: ${walletAddress}`);
+                    break;
+                }
+                case SupportedMethod.FOLLOW_TWITTER_USER: {
+                    const followed = await fireflyBridgeProvider.request(SupportedMethod.FOLLOW_TWITTER_USER, {
+                        id: '952921795316912133',
```
Comment (nuanyang233) [inline src/app/(developers)/components/MethodButton.tsx:64]: ？

File: src/components/Activity/ActivityConnectButton.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                                        },
+                                    );
+                                    onChangeAddress(address);
+                                    refetchActivityClaimCondition();
+                                    return;
+                                }
+                                await AddWalletModalRef.openAndWaitForClose({
+                                    connections: connected,
+                                });
+                                await delay(5000);
```
Comment (nuanyang233) [inline src/components/Activity/ActivityConnectButton.tsx:113]: Is this time too long?

File: src/components/Activity/ActivityMintSuccessDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            </Popover>
+        );
+    }
+
+    return (
+        <Modal open={open} onClose={onClose}>
+            <div className="transform rounded-[12px] bg-primaryBottom transition-all">
+                <div
+                    className="relative inline-flex items-center justify-center gap-2 rounded-t-[12px] p-4 text-center md:h-[56px] md:w-[600px]"
+                    style={{ background: 'var(--m-modal-title-bg)' }}
```
Comment (nuanyang233) [inline src/components/Activity/ActivityMintSuccessDialog.tsx:72]: Why not use Tailwind CSS?

File: src/components/Activity/hooks/useActivityCompose.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                        {
+                            platform_id: '0x01b000',
+                            platform: SourceInURL.Lens,
+                            handle: 'fireflyapp',
+                            name: 'fireflyapp',
+                            hit: true,
+                            score: 0,
+                        },
+                    ],
+                },
```
Comment (nuanyang233) [inline src/components/Activity/hooks/useActivityCompose.tsx:52]: Can extracting to constants？

File: src/providers/telemetry/captureActivityEvent.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    },
+) {
+    runInSafe(() => {
+        const fireflyAccountId = useFireflyStateStore.getState().currentProfileSession?.profileId as string;
+        TelemetryProvider.captureEvent(
+            eventId,
+            {
+                ...params,
+                ...getPublicParameters(eventId, null),
+                firefly_account_id: fireflyAccountId,
```
Comment (guanbinrui) [inline src/providers/telemetry/captureActivityEvent.ts:29]:
```suggestion
```

Comment (guanbinrui) [reply src/providers/telemetry/captureActivityEvent.ts:29]: It's included in the public parameters.

File: src/components/Activity/NavigationBar.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            >
+                <NavigationBarBackIcon width={24} height={24} />
+            </button>
+            <p className="w-full min-w-0 truncate">{children}</p>
+            <button
+                className="h-6 w-6 cursor-pointer"
+                onClick={() => {
+                    captureActivityEvent(EventId.EVENT_SHARE_CLICK, {
+                        firefly_account_id: fireflyAccountId,
+                    });
```
Comment (guanbinrui) [inline src/components/Activity/NavigationBar.tsx:42]:
```suggestion
                    captureActivityEvent(EventId.EVENT_SHARE_CLICK);
```

File: src/app/(activity)/(event)/event/[name]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+                        handle="thefireflyapp"
+                        source={Source.Twitter}
+                        profileId="1583361564479889408"
+                    />
+                    <h2 className="text-base font-semibold leading-6">
+                        <Trans>Step 2 Connect Wallet</Trans>
+                    </h2>
+                    <ActivityConnectCard />
+                    <ActivityPremiumConditionList />
+                    {/* <ActivityRules /> */}
```
Comment (guanbinrui) [inline src/app/(activity)/(event)/event/[name]/page.tsx:97]:
```suggestion
```

File: src/services/getActivityClaimCondition.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    if (authToken) {
+        const response = await fetchJSON<CheckResponse>(url, {
+            headers: {
+                Authorization: `Bearer ${authToken}`,
+            },
+        });
+        return response.data;
+    }
+    const response = await fireflySessionHolder.fetch<CheckResponse>(url);
+    return response.data;
```
Comment (guanbinrui) [inline src/services/getActivityClaimCondition.ts:26]: resolveFireflyResponseData

File: src/modals/AddWalletModal.tsx [hunk 9]
```diff
@@ -31,20 +32,44 @@ export const AddWalletModal = forwardRef<SingletonModalRefCreator<AddWalletModal
...
+                    onBindEvmAddress().then(() => {
+                        onClose();
+                    });
+                    break;
+                case 'solana':
+                    onBindSolanaAddress().then(() => {
+                        onClose();
+                    });
+                    break;
+            }
```
Comment (guanbinrui) [inline src/modals/AddWalletModal.tsx:57]: Move to after `setProps`.

File: src/modals/AddWalletModal.tsx [hunk 10]
```diff
@@ -31,20 +32,44 @@ export const AddWalletModal = forwardRef<SingletonModalRefCreator<AddWalletModal
...
+        }, [platform]);
+
+        useEffect(() => {
+            if (!isConnecting || !account.address) return;
+            onBindEvmAddress();
+        }, [account.address, isConnecting]);
+
```
Comment (guanbinrui) [inline src/modals/AddWalletModal.tsx:64]:
```suggestion
```

File: src/types/bridge.ts [hunk 11]
```diff
@@ -81,8 +92,8 @@ export interface RequestResult {
     [SupportedMethod.GET_WALLET_ADDRESS]: string[];
     [SupportedMethod.CONNECT_WALLET]: string;
     [SupportedMethod.BIND_WALLET]: string; // address
-    [SupportedMethod.IS_TWITTER_USER_FOLLOWING]: boolean;
-    [SupportedMethod.FOLLOW_TWITTER_USER]: boolean;
+    [SupportedMethod.IS_TWITTER_USER_FOLLOWING]: StringifyBoolean;
+    [SupportedMethod.FOLLOW_TWITTER_USER]: StringifyBoolean;
```
Comment (guanbinrui) [inline src/types/bridge.ts:96]: Please stringify booleans in `callNativeMethod`.

File: src/components/Activity/ActivityConnectButton.tsx [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+                            'opacity-0': isRefetching || isLoading,
+                        })}
+                    >
+                        {buttonText}
+                    </span>
+                </Menu.Button>
+                <Menu.Items className="absolute left-1/2 top-[calc(100%+12px)] z-50 flex max-h-[400px] w-[200px] -translate-x-1/2 flex-col overflow-y-auto rounded-[12px] border border-line bg-primaryBottom shadow-lg">
+                    {addresses.map(({ address, ens }) => (
+                        <Menu.Item key={address}>
+                            <a
```
Comment (guanbinrui) [inline src/components/Activity/ActivityConnectButton.tsx:93]: Link

File: src/components/Activity/ActivityConnectButton.tsx [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+                                    });
+                                    refetchActivityClaimCondition();
+                                }}
+                            >
+                                {ens || formatAddress(address, 4)}
+                            </a>
+                        </Menu.Item>
+                    ))}
+                    <Menu.Item>
+                        <a
```
Comment (guanbinrui) [inline src/components/Activity/ActivityConnectButton.tsx:109]: Link

File: src/components/Activity/hooks/useIsFollowTwitterInActivity.ts [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+        queryFn: async () => {
+            if (fireflyBridgeProvider.supported) {
+                return (
+                    (await fireflyBridgeProvider.request(SupportedMethod.IS_TWITTER_USER_FOLLOWING, {
+                        id: profileId,
+                    })) === 'true'
+                );
+            }
+            const profile = await getProfileById(Source.Twitter, profileId);
+            return profile?.viewerContext?.following;
```
Comment (guanbinrui) [inline src/components/Activity/hooks/useIsFollowTwitterInActivity.ts:20]:
```suggestion
            return profile?.viewerContext?.following ?? false;
```

File: src/services/resolveExplorerLink.ts [hunk 15]
```diff
@@ -0,0 +1,21 @@
+import { EVMExplorerResolver } from '@masknet/web3-providers';
+import urlcat from 'urlcat';
+
+const EXPLORER_CONFIG: Record<number, string> = {
+    [84532]: 'https://sepolia.basescan.org',
+};
+
+export function resolveExplorerLink(chainId: number, id: string, type: 'address' | 'tx') {
```
Comment (guanbinrui) [inline src/services/resolveExplorerLink.ts:8]: MV to helpres.

**Pull request merged**